File: 11917 Do Your Own Homework/solution.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        st.nextToken();
        int T = (int)st.nval;
        StringBuilder output = new StringBuilder();
        for(int i = 1; i <= T; ++i) {
            st.nextToken();
            int N = (int)st.nval;
            HashMap<String, Integer> subject = new HashMap<>();
            while(N-- > 0) {
                st.nextToken();
                String name = st.sval;
                st.nextToken();
                subject.put(name, (int)st.nval);
            }
            st.nextToken();
            int days = (int)st.nval;
            st.nextToken();
            if(!subject.containsKey(st.sval))
                output.append("Case ").append(i).append(": Do your own homework!\n");
            else {
                if(days >= subject.get(st.sval))
                    output.append("Case ").append(i).append(": Yesss\n");
                else if(days + 5 < subject.get(st.sval))
                    output.append("Case ").append(i).append(": Do your own homework!\n");
                else
                    output.append("Case ").append(i).append(": Late\n");
            }
        }
        System.out.print(output);
	}
}

File: 11917 Do Your Own Homework/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int T;
    cin >> T;
    for(int i = 1; i <= T; ++i) {
        int N;
        cin >> N;
        unordered_map<string, int> subjects;
        while(N--) {
            string name;
            int days;
            cin >> name >> days;
            subjects[name] = days;
        }
        int due;
        string subject;
        cin >> due >> subject;
        if(!subjects.count(subject))
            cout << "Case " << i << ": Do your own homework!\n";
        else {
            if(due >= subjects[subject])
                cout << "Case " << i << ": Yesss\n";
            else if(due + 5 < subjects[subject])
                cout << "Case " << i << ": Do your own homework!\n";
            else
                cout << "Case " << i << ": Late\n";
        }
    }
    return 0;
}
